services/gateway-api/src/circuit_breaker/circuit_breaker.py

```
@@ -0,0 +1,463 @@
+
+        self._state = CircuitState.CLOSED
+        self._metrics = CircuitMetrics()
+        self._lock = Lock()
+
+        self._opened_at: Optional[float] = None
+        self._backoff_count = 0
+
+        logger.info(
+            f"Circuit breaker '{name}' initialized with "
+            f"failure_threshold={self.config.failure_threshold}, "
+            f"timeout={self.config.timeout_seconds}s"
```

Ruff (0.14.8)

Logging statement uses f-string

(G004)